### PR TITLE
1996 remove current server

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #1996 .

Add any further explanations here.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

**Review by mssalvatore — CHANGES_REQUESTED** [review]
Let's not create a new subpackage (`cc`) yet. When we merge the communication mechanisms, this subpackage won't provide any value.

**Review by mssalvatore — CHANGES_REQUESTED** [review]
I've made some more comments, but the PR is too large and has at least two distinct goals. Split it up into 2 or more PRs, each of which accomplishes a single goal.

**Comment by VakarisZ** [comment]
> Let's not create a new subpackage (`cc`) yet. When we merge the communication mechanisms, this subpackage won't provide any value.

I disagree. It makes it easier to find the files that have the same responsibility until then and it costs nothing to move them back. I'm not sure why you say that it won't provide any value once we merge the `control_client` and `control_channel`, because if we do it in extendable way this package will contain more than one file

File: monkey/tests/unit_tests/infection_monkey/telemetry/conftest.py [hunk 1]
```diff
@@ -11,5 +13,6 @@ def _spy_send_telemetry(telem_category, data):
 
     _spy_send_telemetry.telem_category = None
     _spy_send_telemetry.data = None
-    monkeypatch.setattr(ControlClient, "send_telemetry", _spy_send_telemetry)
+    ControlClient.control_client_object = MagicMock()
+    monkeypatch.setattr(ControlClient.control_client_object, "send_telemetry", _spy_send_telemetry)
```
[the remark this answers is not in the document]
Comment (VakarisZ) [reply monkey/tests/unit_tests/infection_monkey/telemetry/conftest.py:17]: What's the difference? Why do you prefer it that way?

**Comment by VakarisZ** [comment]
Split up the PR. First part is in https://github.com/guardicore/monkey/pull/2016

File: monkey/tests/unit_tests/infection_monkey/telemetry/conftest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilija-lazoroski) [reply monkey/tests/unit_tests/infection_monkey/telemetry/conftest.py:17]: https://stackoverflow.com/a/47101194 Not really some difference between.

File: monkey/infection_monkey/control.py [hunk 2]
```diff
@@ -23,11 +24,17 @@
 PBA_FILE_DOWNLOAD = "https://%s/api/pba/download/%s"
 
 
-class ControlClient(object):
-    proxies = {}
+class ControlClient:
+    # TODO When we have mechanism that support telemetry messenger
+    # with control clients, then this needs to be removed
+    # Ref: infection_monkey.telemetry.base_telem.py
```
[the remark this answers is not in the document]
Comment (mssalvatore) [reply monkey/infection_monkey/control.py:30]: Navigate to a file in GitHub. Click a line. SHIFT+click on a line below to select a block. Click the ellipsis and select "copy permalink"

image

File: monkey/tests/unit_tests/infection_monkey/telemetry/conftest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mssalvatore) [reply monkey/tests/unit_tests/infection_monkey/telemetry/conftest.py:17]: That answer doesn't have anything to do with why I prefer using the MagicMock here.

If you monkey patch the object, then the object still has the capability to do all sorts of things. Monkey patching `send_telemetry()` but leaving all of the other methods intact could result in unexpected behavior down the road if the code decides to try and use any of those methods.

Using a mock object, we're sure it only does the things we've programmed it to do within the confines of what we want to test.

**Comment by mssalvatore** [comment]
>> Let's not create a new subpackage (cc) yet. When we merge the communication mechanisms, this subpackage won't provide any value.

> I disagree. It makes it easier to find the files that have the same responsibility until then and it costs nothing to move them back. I'm not sure why you say that it won't provide any value once we merge the control_client and control_channel, because if we do it in extendable way this package will contain more than one file

@VakarisZ It may contain more than one file, or it may not. As we simplify the API on the Island, the agent should also become simpler. Adding layers in a hierarchy can add difficulty, so unless there's a distinct benefit to do it, it should be avoided.

In this specific case, it introduces a circular dependency between the parent and child packages. `infection_monkey` depends on `infection_monkey.cc` since `main.py` depends on `control.py`. But `infection_monkey.cc.control.py` depends on `infection_monkey.tunnel`, creating a circular dependency between `infection_monkey` and `infection_monkey.cc`.

**Pull request closed without merging**